### ISSUE TITLE
Question:  Cannot eat anybody

### ISSUE THREAD

**Issue by nap82** [huytd/agar.io-clone#413]
I just downloaded and installed it on Ubuntu 15.10, but no matter how large we grow, we cannot eat each other. Is there a setting I need to change?

**Comment by wb9688** [comment]
Does it works when you go back to commit
ef5bdf5f8e61d5806e23db9ef83f8ef7719e0596? If so, it's a bug in my code and
I'll try fixing it...

**Comment by nap82** [comment]
Yes, that worked!  We can now eat blobs!! Thanks for the last good commit #.

**Comment by FireballFalcon** [comment]
Still not working for me.

**Comment by wb9688** [comment]
Try commit ef5bdf5f8e61d5806e23db9ef83f8ef7719e0596. I'll fix it soon and
I've already said that.

**Comment by FireballFalcon** [comment]
Tried it, still doesn't seem to work.

**Comment by wb9688** [comment]
Are you 5 mass bigger than the other player and is the other players mass
more than 10?

**Comment by FireballFalcon** [comment]
Yes

**Comment by wb9688** [comment]
That's weird, can you try the `es6-refactor` branch?

**Comment by FireballFalcon** [comment]
How would I clone that branch?

**Comment by wb9688** [comment]
With `git clone -b es6-refactor https://github.com/huytd/agar.io-clone.git`

**Comment by FireballFalcon** [comment]
Won't install dependencies/start.
`npm install
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/jscs/node_modules/vow-fs/node_modules/glob requires minimatch@'^2.0.1' but will load
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/jscs/node_modules/minimatch,
npm WARN unmet dependency which is version 3.0.0
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/babel-eslint/node_modules/babel-core/node_modules/source-map-support requires source-map@'0.1.32' but will load
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/babel-eslint/node_modules/babel-core/node_modules/source-map,
npm WARN unmet dependency which is version 0.5.6
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/eslint/node_modules/file-entry-cache/node_modules/flat-cache/node_modules/del/node_modules/globby requires glob@'^6.0.1' but will load
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/eslint/node_modules/glob,
npm WARN unmet dependency which is version 5.0.15
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/node-sass/node_modules/node-gyp requires glob@'3 || 4' but will load
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/node-sass/node_modules/glob,
npm WARN unmet dependency which is version 7.0.3
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/node-sass/node_modules/gaze/node_modules/globule/node_modules/glob requires minimatch@'0.3' but will load
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/node-sass/node_modules/gaze/node_modules/globule/node_modules/minimatch,
npm WARN unmet dependency which is version 0.2.14
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/node-sass/node_modules/request/node_modules/http-signature/node_modules/sshpk requires assert-plus@'^1.0.0' but will load
npm WARN unmet dependency /home/ubuntu/workspace/agar.io-clone/node_modules/node-sass/node_modules/request/node_modules/http-signature/node_modules/assert-plus,
npm WARN unmet dependency which is version 0.2.0
zanderl:~/workspace/agar.io-clone (es6-refactor) $ npm start

> agar-clone@1.0.0 start /home/ubuntu/workspace/agar.io-clone
> babel-node server/index

sh: 1: babel-node: not found

npm ERR! Linux 4.2.0-c9
npm ERR! argv "/home/ubuntu/.nvm/versions/node/v4.4.3/bin/node" "/home/ubuntu/.nvm/versions/node/v4.4.3/bin/npm" "start"
npm ERR! node v4.4.3
npm ERR! npm  v2.15.1
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! agar-clone@1.0.0 start: `babel-node server/index`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the agar-clone@1.0.0 start script 'babel-node server/index'.
npm ERR! This is most likely a problem with the agar-clone package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     babel-node server/index
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs agar-clone
npm ERR! Or if that isn't available, you can get their info via:
npm ERR! 
npm ERR!     npm owner ls agar-clone
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/ubuntu/workspace/agar.io-clone/npm-debug.log`

**Comment by dynasty4544** [comment]
Eating won't work for me as well

**Comment by geopi** [comment]
Making a heavily modified version i stumbled into that bug too.

The bug I had was related to the the quadtree used for collision detection.

There is a temporary fix until a solution is found :

In server.js remove this : 
```
 tree.clear();
        users.forEach(tree.put);
        var playerCollisions = [];

        var otherUsers =  tree.get(currentPlayer, check);

        playerCollisions.forEach(collisionCheck);
```

and put this instead : 
```
        var playerCollisions = [];
         users.forEach(check);
        playerCollisions.forEach(collisionCheck);
```

It basically forces the detection in the whole map

Its dirty, I know.

**Comment by smarty6788** [comment]
I've found a fix but it includes removing the split

go to config.json and change this  "limitSplit": 1,

**Comment by johnsmitherson** [comment]
The problem I have is that smaller player is eaten only if it is in the center of bigger player, no matter what the size different. Can someone show what to change to make it that smaller player is eaten as soon it is completely under bigger player?

**Comment by aspriddell** [comment]
I made a sort of fix by finding the code in server.js (line 538) and removed the multipliers. As far as I know it is working much better.
